File: November 2022/17 November 2022/Java/Count of Subarrays.java
//{ Driver Code Starts
// Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    // Driver code
    public static void main(String[] args) throws Exception {
        BufferedReader br =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim());
        while (t-- > 0) {
            String[] str = br.readLine().trim().split(" ");
            int n = Integer.parseInt(str[0]);
            int k = Integer.parseInt(str[1]);
            str = br.readLine().trim().split(" ");
            int[] arr = new int[n];
            for (int i = 0; i < n; i++) {
                arr[i] = Integer.parseInt(str[i]);
            }

            long ans = new Solution().countSubarray(arr, n, k);

            System.out.println(ans);
        }
    }
}

// } Driver Code Ends


// User function Template for Java

class Solution {

    long countSubarray(int arr[], int n, int k) {
        int s = 0;
        int i = 0;
        while(i < n) {
            int ct = 0;
            if (arr[i] > k) {
                i++;
                continue;
            }
            while (i < n && arr[i] <= k) {
                i++;
                ct++;
            }
            s += (long)(((long)ct * (long)(ct + 1)) / 2);
        }
        return (long)(((long)n * (long)(n + 1)) / 2 - s);
    }
}

File: November 2022/17 November 2022/Python/Count of Subarrays.py
#User function Template for python3
class Solution:


	def countSubarray(self,arr, n, k):
        dp = [0 for i in range(n)]
        for i in range(n):
            if(arr[i] > k):
                dp[i] = n-i
        tmp = 0
        for i in range(n-1, -1, -1):
            if(dp[i] != 0):
                tmp = dp[i]
                continue
            dp[i] = tmp
        return sum(dp)

#{ 
 # Driver Code Starts
#Initial Template for Python 3


# Driver code 
if __name__ == "__main__": 		
    tc=int(input())
    while tc > 0:
        n, k=map(int, input().strip().split())
        arr=list(map(int, input().strip().split()))
        ob = Solution()
        ans=ob.countSubarray(arr, n, k)
        print(ans)
        tc=tc-1
# } Driver Code Ends